cmd/infra/aws/create_iam.go

```diff
@@ -155,7 +169,17 @@ func (o *CreateIAMOptions) CreateIAM(ctx context.Context, client crclient.Client
 		return nil, err
 	}
 
-	awsSession := awsutil.NewSession("cli-create-iam", o.AWSCredentialsFile, o.AWSKey, o.AWSSecretKey, o.Region)
+	var awsSession *session.Session
```

Comment:
The sheer number of places where you need to add these new flags and validation is too high to be tenable. Let's break out some

```go
type AWSCredentialOptions struct {
    RoleARN string
    STSCreds string
    // whatever else
}

func DefaultOptions() *AWSCredentialOptions {
    return &AWSCredentialOptions{/* place any meaningful defaults here */}
}

func BindOptions(opts *AWSCredentialOptions, flags *pflag.FlagSet) {
    flags.StringVar(&opts.RoleArn, "role-arn", opts.RoleArn, "The ARN of the role to assume when creating the iam.")
    flags.StringVar(&opts.StsCredentialsFile, "sts-creds", opts.StsCredentialsFile, "Path to STS credentials file to use when assuming the role.")
}

func (opts *AWSCredentialOptions) Validate() error {
    /* add validation here */
}
```

Then, embed this struct into the others where needed and call `BindOptions()` on it where flags are being added.

Reply:
@stevekuznetsov because of the --secret-creds flag we sometimes need the roleArn without the need of stsCreds. I have created a new util validation function for validating the required flags here cmd/util/credentials_validation.go

Reply:
Let's find a factoring where we bind the flags once in the repo and validate them in one flow. If that means that the options need parameters on creation and/or validation, so be it. For all commands that want to authenticate to AWS, let's provide a package that lets them bind the flags they need and get an AWS session out the other end. A good litmus test IMHO will be that this PR adds only one line that binds `sts-creds` as a flag to a struct member.

Reply:
@stevekuznetsov Just reading over the PR and I agree with you but since I'm leaving for PTO today and this feature is time sensitive I think this would be more of nice to have in this scenario. @muraee I don't know if you have the time to look into this approach or maybe this can be a separate Jira in our backlog?

Reply:
I can't speak to timelines but I think trending in the direction I outline here will help us immensely with maintenance of the codebase.

Reply:
@Patryk-Stefanski given that you'll be out, is it ok if we get someone else to take over the PR after today? I don't think @stevekuznetsov's are that big, but agree they would help with the maintenance of the code.

Reply:
@csrwng I have added @muraee as contributor and I can add anyone else aswell just let me know.